File: pubsub/src/main/java/glitch/pubsub/events/BitsEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import glitch.pubsub.object.json.BitsMessage;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class BitsEvent extends AbstractMessageEvent<BitsMessage> implements IEvent<GlitchPubSub> {
    public BitsEvent(GlitchPubSub client, Topic topic, BitsMessage data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/BitsEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import glitch.pubsub.`object`.json.BitsMessage

class BitsEvent(client: GlitchPubSub, topic: Topic, data: BitsMessage) : AbstractMessageEvent<BitsMessage>(client, topic, data), IEvent<GlitchPubSub>
